src/secret_connection.rs

```diff
@@ -393,16 +413,10 @@ fn hash24(input: &[u8]) -> [u8; 24] {
     return res;
 }
 
-// increment nonce big-endian by 2 with wraparound.
-fn incr2_nonce(nonce: &mut [u8; 24]) {
-    incr_nonce(nonce);
-    incr_nonce(nonce);
-}
-
 // TODO: Check if internal representation is big or small endian
 // increment nonce big-endian by 2 with wraparound.
-fn incr_nonce(nonce: &mut [u8; 24]) {
-    for i in (0..24).rev() {
+fn incr_nonce(nonce: &mut [u8; 12]) {
+    for i in (0..12).rev() {
```

Comment:
I'd suggest using the [byteorder crate](https://github.com/BurntSushi/byteorder) for this, and `checked_add` for the addition operation below.

Also I'd strongly suggest little endian, among other reasons because ChaCha20 is little endian internally.

Reply:
> among other reasons because ChaCha20 is little endian internally.

did not know that. I think it makes sense to make this little endian then. 

> I'd suggest using the byteorder crate for this, and checked_add for the addition operation below.

Sounds good to me